### PR TITLE
Add OutBound Type to aks_config_v2

### PR DESCRIPTION
**Pull request by mjura**
Issue: https://github.com/rancher/aks-operator/issues/721

### CODE REVIEW

File: rancher2/schema_cluster_aks_config_v2.go [hunk 1]
```diff
@@ -2,13 +2,18 @@ package rancher2
...
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
 )
 
 const (
 	clusterAKSV2Kind   = "aksV2"
 	clusterDriverAKSV2 = "AKS"
 )
 
+var (
+	clusterAKSOutboundType = []string{"loadBalancer", "managedNATGateway", "userAssignedNATGateway", "userDefinedRouting"}
```
Comment (yiannistri) [inline rancher2/schema_cluster_aks_config_v2.go]: Optional suggestion: Not sure if these are case sensitive (probably not) but maybe we should match these to how they appear in the [Azure docs](https://learn.microsoft.com/en-us/azure/aks/egress-outboundtype#outbound-types-in-aks) i.e. `managedNatGateway` and `userAssignedNatGateway`.

File: rancher2/schema_cluster_aks_config_v2.go [hunk 2]
```diff
@@ -263,6 +268,13 @@ func clusterAKSConfigV2Fields() map[string]*schema.Schema {
 			Computed:    true,
 			Description: "The AKS node resource group name",
 		},
+		"outboung_type": {
+			Type:         schema.TypeString,
+			Optional:     true,
+			Default:      "loadBalancer",
+			Description:  "The AKS outbound type for the egress traffic",
+			ValidateFunc: validation.StringInSlice(clusterAKSOutboundType, true),
```
Comment (yiannistri) [inline rancher2/schema_cluster_aks_config_v2.go:276]: TIL about `validation.StringInSlice` :+1:

**Review by yiannistri — APPROVED** [review]

File: rancher2/schema_cluster_aks_config_v2.go (hunk 2 above)
Comment (mjura) [reply rancher2/schema_cluster_aks_config_v2.go:276]: validation function has already non case sensitive flag enabled

File: rancher2/schema_cluster_aks_config_v2.go (hunk 1 above)
Comment (mjura) [reply rancher2/schema_cluster_aks_config_v2.go]: it is `non case sensitive`

**Review by matttrach — APPROVED** [review]

**Pull request merged**